sdk/metric/meter_test.go

```diff
@@ -474,3 +475,321 @@ func TestMetersProvideScope(t *testing.T) {
 	assert.NoError(t, err)
 	metricdatatest.AssertEqual(t, want, got, metricdatatest.IgnoreTimestamp())
 }
+
+func TestAttributeFilter(t *testing.T) {
+	one := 1.0
+	two := 2.0
+	testcases := []struct {
+		name       string
+		register   func(t *testing.T, mtr metric.Meter) error
+		wantMetric metricdata.Metrics
+	}{
+		{
+			name: "AsyncFloat64Counter",
+			register: func(t *testing.T, mtr metric.Meter) error {
+				ctr, err := mtr.AsyncFloat64().Counter("afcounter")
+				if err != nil {
+					return err
+				}
+				return mtr.RegisterCallback([]instrument.Asynchronous{ctr}, func(ctx context.Context) {
+					ctr.Observe(ctx, 1.0, attribute.Int("version", 1))
+					ctr.Observe(ctx, 2.0, attribute.Int("version", 2))
+				})
+			},
+			wantMetric: metricdata.Metrics{
+				Name: "afcounter",
+				Data: metricdata.Sum[float64]{
+					DataPoints: []metricdata.DataPoint[float64]{
+						{Value: 2.0}, // TODO: This should be 3.0.  It is a bug in the sum aggregator
```

Comment:
Please open an issue to track this bug/todo and update the comment with the issue link.